src/Symfony/Component/Lock/Store/MemcachedStore.php

```diff
@@ -52,6 +58,127 @@ public function __construct(\Memcached $memcached, $initialTtl = 300)
+                }
+                $opt = constant('Memcached::OPT_'.$name);
+
+                unset($options[$name]);
+                $options[$opt] = $value;
+            }
+            $client->setOptions($options);
+
+            // parse any DSN in $servers
+            $params = preg_replace_callback('#^memcached://(?:([^@]*+)@)?#', function ($m) use (&$username, &$password) {
+                if (!empty($m[1])) {
+                    list($username, $password) = explode(':', $m[1], 2) + array(1 => null);
+                }
+
+                return 'file://';
+            }, $dsn);
+            if (false === $params = parse_url($params)) {
+                throw new InvalidArgumentException(sprintf('Invalid Memcached DSN: %s', $dsn));
+            }
+            if (!isset($params['host']) && !isset($params['path'])) {
+                throw new InvalidArgumentException(sprintf('Invalid Memcached DSN: %s', $dsn));
+            }
+            if (isset($params['path']) && preg_match('#/(\d+)$#', $params['path'], $m)) {
+                $params['weight'] = $m[1];
+                $params['path'] = substr($params['path'], 0, -strlen($m[0]));
+            }
+            $params += array(
+                'host' => isset($params['host']) ? $params['host'] : $params['path'],
+                'port' => isset($params['host']) ? 11211 : null,
+                'weight' => 0,
+            );
+            if (isset($params['query'])) {
+                parse_str($params['query'], $query);
+                $params += $query;
+            }
+
+            $servers = array(array($params['host'], $params['port'], $params['weight']));
+
+            // set client's servers, taking care of persistent connections
+            if (!$client->isPristine()) {
+                $oldServers = array();
+                foreach ($client->getServerList() as $server) {
+                    $oldServers[] = array($server['host'], $server['port']);
+                }
+
+                $newServers = array();
+                foreach ($servers as $server) {
+                    if (1 < count($server)) {
+                        $server = array_values($server);
+                        unset($server[2]);
+                        $server[1] = (int) $server[1];
+                    }
+                    $newServers[] = $server;
+                }
+
+                if ($oldServers !== $newServers) {
+                    // before resetting, ensure $servers is valid
+                    $client->addServers($servers);
+                    $client->resetServerList();
+                }
+            }
+            $client->addServers($servers);
+
+            if (null !== $username || null !== $password) {
+                if (!method_exists($client, 'setSaslAuthData')) {
+                    trigger_error('Missing SASL support: the memcached extension must be compiled with --enable-memcached-sasl.');
```

Comment:
What about a `E_USER_WARNING`? As we're clearly going to call an undefined method.